lib/rules/newline-after-var.js

```diff
@@ -64,6 +64,27 @@ module.exports = function(context) {
     }
 
     /**
+     * Determine if provided nodeType is a function specifier
+     * @private
+     * @param {string} nodeType - nodeType to test
+     * @returns {boolean} True if `nodeType` is a function specifier
+     */
+    function isFunctionSpecifier(nodeType) {
+        return nodeType === "FunctionDeclaration" || nodeType === "FunctionExpression" ||
+            nodeType === "ArrowFunctionExpression";
+    }
+
+    /**
+     * Determine if provided node is the last of his parent
+     * @private
+     * @param {ASTNode} node - node to test
+     * @returns {boolean} True if `node` is last of his parent
+     */
+    function isLastNode(node) {
+        return node.parent.body.pop() === node;
```

Comment:
This changes what's in body, so it will break other rules. You should just check the node at node.body.length - 1.